### PR TITLE
Add constants and MERCINFO_GID to *getmercinfo script command

### PR DESCRIPTION
**Pull request by AnnieRuru**
[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style
### Pull Request Prelude
- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Issues addressed
while fixing https://github.com/HerculesWS/Hercules/pull/2395 and I need a script command to return mercenary GID,
so I copy existing code from rAthena

### Changes Proposed
Add constants and MERCINFO_GID to *getmercinfo script command

### Tested with
```c
prontera,155,185,5	script	kjdshfksdf	1_F_MARIA,{
	getitem Bow_Mercenary_Scroll1, 1;
	
	for ( .@i = 0; .@i <= 8; ++.@i )
		mes getmercinfo(.@i) +"";
	
	mes rid2name(getmercinfo(MERCINFO_GID));
	mes getunitname(getmercinfo(MERCINFO_GID)); // return nullpo
	close;
}
```

### Affected Branches
* Master

### Known Issues and TODO List
none

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2397)

File: src/map/script.c [hunk 1]
```diff
@@ -15617,16 +15617,9 @@ static BUILDIN(gethominfo)
...
 	struct map_session_data* sd;
-	struct mercenary_data* md;
-
-	type = script_getnum(st,2);
-
 	if (script_hasdata(st,3)) {
-		int char_id = script_getnum(st,3);
-
-		if ((sd = script->charid2sd(st, char_id)) == NULL) {
+		if ((sd = script->charid2sd(st, script_getnum(st,3))) == NULL) {
```
Comment (Asheraf) [inline src/map/script.c]: missing a space after the `,` also please check the reset of the line you changed for style issues

File: src/map/script.c [hunk 2]
```diff
@@ -15635,27 +15628,30 @@ static BUILDIN(getmercinfo)
 			return true;
 	}
 
-	md = ( sd->status.mer_id && sd->md ) ? sd->md : NULL;
+	struct mercenary_data *md = (sd->status.mer_id && sd->md)? sd->md : NULL;
+	int type = script_getnum(st,2);
+	if (md == NULL) {
+		if (type == 2)
```
Comment (Asheraf) [inline src/map/script.c]: `MERCINFO_NAME`

**Comment by AnnieRuru** [comment]
fix line break coding style in switch cases

**Pull request merged**